includes/admin/upgrades/upgrade-functions.php

```diff
@@ -1550,13 +1513,13 @@ function edd_upgrade_render_v30_migration() {
 						<?php wp_nonce_field( 'edd_ajax_export', 'edd_ajax_export' ); ?>
 
 						<?php if ( ! $customer_addresses_complete ) : ?>
-							<span class="edd-migration allowed" style="<?php echo ! $migration_complete ? '' : 'display: none'; ?>">
+							<span class="edd-migration allowed" style="<?php echo $migration_complete ? '' : 'display: none'; ?>">
 								<input type="submit" id="migrate-customer-addresses-submit" value="<?php esc_attr_e( 'Upgrade Database', 'easy-digital-downloads' ); ?>" class="button-primary"/>
 							</span>
 
-							<span class="edd-migration unavailable" style="<?php echo $migration_complete ? '' : 'display: none'; ?>">
+							<span class="edd-migration unavailable" style="<?php echo ! $migration_complete ? '' : 'display: none'; ?>">
 								<input type="submit" disabled id="migrate-customer-addresses-submit" value="<?php esc_attr_e( 'Upgrade Database', 'easy-digital-downloads' ); ?>" class="button-secondary"/>
-								&mdash; <?php esc_html_e( 'Your customer addresses been upgraded.', 'easy-digital-downloads' ); ?>
+								&mdash; <?php esc_html_e( 'Please complete the previous steps before updating customer addresses.', 'easy-digital-downloads' ); ?>
```

Comment:
I ran the migration via the UI and once it was done, all of these "Please complete the previous steps..." messages were displaying. If the migration step has completed successfully, this message should disappear or be replaced with something like "Migration step complete!".

So maybe the UI needs to have three possible statuses:
- unavailable (not ready)
- allowed
- complete

With messages shown/hidden accordingly.

Reply:
@robincornett I slightly addressed that in https://github.com/easydigitaldownloads/easy-digital-downloads/compare/issue/6810...issue/6810-2 by not showing the "unavailable" message and instead disabling the input inside of "allowed".

I think a fair amount will still change if we have time in https://github.com/easydigitaldownloads/easy-digital-downloads/issues/6880 so I'm not sure making too many changes in this PR beyond the initial interaction blocking is necessary.